pkg/reconciler/v1alpha1/taskrun/resources/pod_test.go

```diff
@@ -435,3 +420,42 @@ func TestMakeWorkingDirScript(t *testing.T) {
 		})
 	}
 }
+
+func TestAddReadyAnnotation(t *testing.T) {
+	type testcase struct {
+		desc                       string
+		pod                        *corev1.Pod
+		updateFunc                 UpdatePod
+		expectedAnnotationsContain map[string]string
+		expectedErr                error
+	}
+	testerror := xerrors.New("error updating pod")
+	for _, c := range []testcase{{
+		desc:                       "missing ready annotation is added to provided pod",
+		pod:                        &corev1.Pod{},
+		updateFunc:                 func(p *corev1.Pod) (*corev1.Pod, error) { return p, nil },
+		expectedAnnotationsContain: map[string]string{ReadyAnnotation: readyAnnotationValue},
+		expectedErr:                nil,
+	}, {
+		desc:                       "errors experienced during update are returned",
+		pod:                        &corev1.Pod{},
+		updateFunc:                 func(p *corev1.Pod) (*corev1.Pod, error) { return p, testerror },
+		expectedAnnotationsContain: map[string]string{ReadyAnnotation: readyAnnotationValue},
+		expectedErr:                testerror,
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			err := AddReadyAnnotation(c.pod, c.updateFunc)
+			if err != c.expectedErr {
+				t.Errorf("expected error %v but received %v", c.expectedErr, err)
+			}
+			for a, v := range c.expectedAnnotationsContain {
```

Comment:
maybe we can avoid looping here since the only test cases we have only have one annotation?

Reply:
Done, I've removed the hash map looping and just check the single expected annotation.